win32ss/user/ntuser/defwnd.c

```diff
@@ -1068,6 +1068,29 @@ IntDefWindowProc(
          }
          break;
 
+      case WM_CREATE:
+      {
+         RECT Rect;
+         HBRUSH hBrush = Wnd->pcls->hbrBackground;
+         if (!hBrush) return 0;
+         if (hBrush <= (HBRUSH)COLOR_MENUBAR)
+         {
+            hBrush = IntGetSysColorBrush(HandleToUlong(hBrush));
+         }
+         if (Wnd->pcls->style & CS_PARENTDC)
+         {
+            /* can't use GetClipBox with a parent DC or we fill the whole parent */
+            IntGetClientRect(Wnd, &Rect);
+            GreDPtoLP((HDC)wParam, (LPPOINT)&Rect, 2);
+         }
+         else
+         {
+            GdiGetClipBox((HDC)wParam, &Rect);
+         }
+         FillRect((HDC)wParam, &Rect, hBrush);
+         return (0);
```

Comment:
There is no need to duplicate the whole case code: simply conditionally return the correct value(s).

Reply:
Yeah, that's what I just wanted to point out too.

Reply:
Sure, I merged the code into one block, it'll make more sense and clarity in the end.